### PR TITLE
add eslint to client and fix react test

### PR DESCRIPTION
**Pull request by jd2rogers2**
## Dev Summary
- re-configuring eslint to be helpful
  - was previously giving us errors in all of the `build` files which made the output too hard to read
- fixes a bunch of missing/extra semis
- adding a line to the pull request template to enforce running lint and test commands
- fixes our react app.test to check for our logo rather than text that isn't there (was breaking before)

## Test Plan

1. `cd client && npm run checks`
see that there are no eslint errors or jest errors
2. remove a `;` from any component (i.e. `client/src/AboutUs.tsx`) and run `npm run checks` again
see that we get an eslint error and therefore eslint is working

## Resources

- https://www.youtube.com/watch?v=bfyI9yl3qfE
- https://eslint.org/docs/rules/

### CODE REVIEW

File: client/src/Header.tsx [hunk 1]
```diff
@@ -73,7 +73,6 @@ function Header() {
                     </NavLink>
                     <NavLink className={classes.navLink} to="/about_us">About Us</NavLink>
                     <NavLink className={classes.navLink} to="/how_it_works">How It Works</NavLink>
-                    <NavLink className={classes.navLink} to="/library_and_forum">Library & Forum</NavLink>
```
Comment (jd2rogers2) [inline client/src/Header.tsx:76]: we don't have this page any more, not sure how this piece stuck around

File: server/src/assets/assets.controller.ts [hunk 2]
```diff
@@ -14,8 +14,8 @@ import {
 import { AssetsService } from './assets.service';
 import { CreateAssetDto } from './dto/create-asset.dto';
 import { UpdateAssetDto } from './dto/update-asset.dto';
+// authGaurd excluded because it is outside of the scope of this PR
 // import { JwtAuthGuard } from '../auth/guards/jwt-auth.guard';
-//authGaurd excluded because it is outside of the scope of this PR
```
Comment (jd2rogers2) [inline server/src/assets/assets.controller.ts:18]: comments go above the line they're referring to

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -26,6 +26,13 @@ You should include screenshots to help the reviewer understand what to look for
 changes are includes. GIFs help immensely when explaining interactions and animations.
 -->
 
+preReqs:
+
+[ ] following command succeeds `cd client && npm run checks && cd ../server && npm run checks`
```
Comment (jd2rogers2) [inline .github/pull_request_template.md:31]: server side scripts, eslint config, etc. will come in a follow up PR
but doing here so i don't forget to add later

File: client/src/types/material-ui.d.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (yndajas) [inline client/src/types/material-ui.d.ts:1]: Think you can delete this? Both imports are used

**Review by yndajas — CHANGES_REQUESTED** [review]
- One line that can probably be removed
- Might be obvious, but `npm run checks` failed until I ran `npm start` - not sure if the client needs to be running or if I just need to have run it at least once, but might be good to include that in the testing details

File: client/src/types/material-ui.d.ts (hunk 4 above)
Comment (jd2rogers2) [reply client/src/types/material-ui.d.ts:1]: they're actually not, we are creating our own types with the same names
(which i'm not sure if this is a good thing, but a bunch of other things break if we remove this module declaration or if we remove these imports 🤷, fixing either of those will have to come in another PR)

if you remove my comment and run `npm run lint` from client folder, you should see warnings. lmk if not

**Comment by jd2rogers2** [comment]
> Might be obvious, but npm run checks failed until I ran npm start - not sure if the client needs to be running or if I just need to have run it at least once, but might be good to include that in the testing details

hmmm, no idea why that's the case. i don't think that should happen
in fact, i didn't run `start` until after all of my changes as an after thought

**Comment by yndajas** [comment]
> > Might be obvious, but npm run checks failed until I ran npm start - not sure if the client needs to be running or if I just need to have run it at least once, but might be good to include that in the testing details
> 
> hmmm, no idea why that's the case. i don't think that should happen in fact, i didn't run `start` until after all of my changes as an after thought

Strange - it's working now without the client running, but this is after having run the client at least once. It said something about missing a checks folder I think, which doesn't quite make sense to me

File: client/src/types/material-ui.d.ts (hunk 4 above)
Comment (yndajas) [reply client/src/types/material-ui.d.ts:1]: Yep, that gave warnings. Interesting!

**Review by yndajas — APPROVED** [review]

**Pull request merged**